Modules/Microsoft365DSC/DSCResources/MSFT_EXOAuthenticationPolicy/MSFT_EXOAuthenticationPolicy.psm1

```diff
@@ -329,7 +351,25 @@ function Set-TargetResource
     {
         Write-Verbose -Message "Authentication Policy '$($Identity)' exists. Updating settings."
         Set-AuthenticationPolicy -Identity $Identity @NewAuthenticationPolicyParams | Out-Null
+        $NeedToCheckAssignedUsers = $true
     }
+
+    #region Update Assigned Users
```

Comment:
At the moment it is not possible to set AuthenticationPolicy attribute of a user to $null (default value) again after a policy was assigned.